lib/cocoapods-core/specification/dsl.rb

```diff
@@ -1316,6 +1316,28 @@ def subspec(name, &block)
         subspec
       end
 
+      # Represents a test specification for the library. Here you can place all
+      # your unit tests for your podspec along with the test dependencies.
+      #
+      # ---
+      #
+      # @example
+      #
+      #   Pod::Spec.new do |spec|
+      #     spec.name = 'NSAttributedString+CCLFormat'
+      #
+      #     spec.test_spec do |test_spec|
+      #       test_spec.source_files = 'NSAttributedString+CCLFormatTests.m'
+      #       test_spec.dependency 'Expecta'
+      #     end
+      #   end
+      #
+      def test_spec(&block)
+        subspec = Specification.new(self, 'Tests', true, &block)
```

Comment:
To avoid future DSL gotchas: it's worth considering about making sure a test target can be configured to support multiple test targets, for example you might have unit tests + uiintegration tests (we do in Eigen, but we don't in any of our libs FWIW)

You might not have to do the work today, but building the DSL in a way that supports that is 👍